client/state/data-layer/wpcom-http/index.js

```diff
@@ -26,6 +27,7 @@ const fetcherMap = method => get( {
 export const successMeta = data => ( { meta: { dataLayer: { data } } } );
 export const failureMeta = error => ( { meta: { dataLayer: { error } } } );
 export const progressMeta = ( { size: total, loaded } ) => ( { meta: { dataLayer: { progress: { total, loaded } } } } );
+export const requestKey = ( { path, query } ) => JSON.stringify( { type: WPCOM_HTTP_REQUEST, path, query } );
```

Comment:
where action.query is the deterministically serialized version of the query?

wpcom _may_ support arrays and objects in queries too. need to check.